src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy

```diff
@@ -504,6 +538,31 @@ public class GenerateProtoTask extends DefaultTask {
     }
   }
 
+  /**
+   * Used to expose inputs to Gradle, not to be called directly.
+   */
+  @Optional
+  @Nested
+  protected DescriptorSetOptions getDescriptorSetOptions() {
```

Comment:
Why is it not to be called directly? This is the public accessor for DescriptorSetOptions used in the example from README.

It seems that you cannot make it throw when `generateDescriptorSet == false` as I requested earlier, because Gradle will query this property unconditionally. This is a bit unfortunate as Gradle doesn't have an annotation for "this is public, but don't use it for up-to-date checks".

Reply:
I'm not sure what you mean by `throw when generateDescriptorSet == false`. I reverted to the origin public property being `@Internal` (that's Gradle's annotation for "this is public, but don't use it for up-to-date checks").

Reply:
If we made accessing `descriptorSetOptions` throw when `generateDescriptorSet == false`, the user would get more meaningful error message (see `getDescriptorPath()`), compared to getting `null` for `descriptorSetOptions` and then NPE down the road.

I find `@Internal` on a public property to be confusing, especially when you add the auxiliary properties like `getDescriptorSetOptionsInternal()` that also have "Internal" in the name. Can you rename them to `*ForUpToDateCheck()`?